pyproject.toml

```diff
@@ -25,6 +25,9 @@ jupyterlab = "^4.0.5"
 python = "^3.8"
 ruff = "^0.0.284"
 
+[tool.poetry.dev-dependencies]
```

Comment:
`dev-dependencies` is obsolete, we should replace this by the dev dependency group: https://python-poetry.org/docs/managing-dependencies#dependency-groups

And may the dev dependency group optional. And include this optional group in the `poetry install`in the CI.